app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -905,6 +907,99 @@ class StateFragmentTest {
     }
   }
 
+  @Test
+  fun testStateFragment_forTwoWrongAnswers_revealHint() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_0).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      clickContinueInteractionButton()
+      clickContinueInteractionButton()
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+      selectMultipleChoiceOption(optionPosition = 1)
+      clickContinueNavigationButton()
+      selectMultipleChoiceOption(optionPosition = 1)
+      clickContinueNavigationButton()
+      onView(withId(R.id.hints_and_solution_fragment_container)).check(
+        matches(
+          not(isDisplayed())
+        )
+      )
+      selectMultipleChoiceOption(optionPosition = 1)
+      selectMultipleChoiceOption(optionPosition = 1)
+      onView(withId(R.id.hints_and_solution_fragment_container)).check(
+        matches(
+          isDisplayed()
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_forTwoWrongAnswers_revealHint_dotDisappears() {
+    launchForExploration(FRACTIONS_EXPLORATION_ID_0).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      clickContinueInteractionButton()
+      clickContinueInteractionButton()
+      selectMultipleChoiceOption(optionPosition = 2)
+      clickContinueNavigationButton()
+      selectMultipleChoiceOption(optionPosition = 1)
+      clickContinueNavigationButton()
+      selectMultipleChoiceOption(optionPosition = 1)
+      clickContinueNavigationButton()
+
+      selectMultipleChoiceOption(optionPosition = 1)
+      selectMultipleChoiceOption(optionPosition = 1)
+      onView(withId(R.id.dot_hint)).check(
+        matches(
+          isDisplayed()
+        )
+      )
+      onView(withId(R.id.hints_and_solution_fragment_container)).perform(click())
+      onView(withId(R.id.reveal_hint_button)).perform(click())
+      onView(isRoot()).perform(pressBack())
+      onView(withId(R.id.dot_hint)).check(
+        matches(
+          not(isDisplayed())
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_forTwoWrongAnswers_revealHint_configurationChange_newHintNotRevealed() {
```

Comment:
Can you please verify this fails correctly without the fix in place?

Reply:
Ah I forgot to remove these tests. I added them when my previous implementation worked.
Also these tests won't be working here as there is a dependency `HintsAndSolutionConfigFastShowTestModule`
Which will be showing the hint without any interaction